### PR TITLE
ensure control-plane-operator and hypershift operator are nonroot

### PR DESCRIPTION
**Pull request by relyt0925**

### CODE REVIEW

**Comment by relyt0925** [comment]
for reference
```
Tylers-MacBook-Pro:hypershift tylerlisowski$ kubectl get pods -n hypershift -o yaml
apiVersion: v1
items:
- apiVersion: v1
  kind: Pod
  metadata:
    annotations:
      cni.projectcalico.org/podIP: 172.30.64.142/32
      cni.projectcalico.org/podIPs: 172.30.64.142/32
      k8s.v1.cni.cncf.io/network-status: |-
        [{
            "name": "",
            "ips": [
                "172.30.64.142"
            ],
            "default": true,
            "dns": {}
        }]
      k8s.v1.cni.cncf.io/networks-status: |-
        [{
            "name": "",
            "ips": [
                "172.30.64.142"
            ],
            "default": true,
            "dns": {}
        }]
      openshift.io/scc: anyuid
    creationTimestamp: "2021-07-02T05:13:14Z"
    generateName: operator-76898bcfc5-
    labels:
      name: operator
      pod-template-hash: 76898bcfc5
    managedFields:
    - apiVersion: v1
      fieldsType: FieldsV1
      fieldsV1:
        f:metadata:
          f:generateName: {}
          f:labels:
            .: {}
            f:name: {}
            f:pod-template-hash: {}
          f:ownerReferences:
            .: {}
            k:{"uid":"8f74e8fa-d167-4d54-a2ce-a49ae6e379f4"}:
              .: {}
              f:apiVersion: {}
              f:blockOwnerDeletion: {}
              f:controller: {}
              f:kind: {}
              f:name: {}
              f:uid: {}
        f:spec:
          f:containers:
            k:{"name":"operator"}:
              .: {}
              f:args: {}
              f:command: {}
              f:env:
                .: {}
                k:{"name":"MY_NAMESPACE"}:
                  .: {}
                  f:name: {}
                  f:valueFrom:
                    .: {}
                    f:fieldRef:
                      .: {}
                      f:apiVersion: {}
                      f:fieldPath: {}
              f:image: {}
              f:imagePullPolicy: {}
              f:name: {}
              f:ports:
                .: {}
                k:{"containerPort":9000,"protocol":"TCP"}:
                  .: {}
                  f:containerPort: {}
                  f:name: {}
                  f:protocol: {}
              f:resources: {}
              f:terminationMessagePath: {}
              f:terminationMessagePolicy: {}
          f:dnsPolicy: {}
          f:enableServiceLinks: {}
          f:restartPolicy: {}
          f:schedulerName: {}
          f:securityContext:
            .: {}
            f:seLinuxOptions:
              f:level: {}
          f:serviceAccount: {}
          f:serviceAccountName: {}
          f:terminationGracePeriodSeconds: {}
      manager: kube-controller-manager
      operation: Update
      time: "2021-07-02T05:13:14Z"
    - apiVersion: v1
      fieldsType: FieldsV1
      fieldsV1:
        f:metadata:
          f:annotations:
            f:cni.projectcalico.org/podIP: {}
            f:cni.projectcalico.org/podIPs: {}
      manager: calico
      operation: Update
      time: "2021-07-02T05:13:15Z"
    - apiVersion: v1
      fieldsType: FieldsV1
      fieldsV1:
        f:metadata:
          f:annotations:
            f:k8s.v1.cni.cncf.io/network-status: {}
            f:k8s.v1.cni.cncf.io/networks-status: {}
      manager: multus
      operation: Update
      time: "2021-07-02T05:13:15Z"
    - apiVersion: v1
      fieldsType: FieldsV1
      fieldsV1:
        f:status:
          f:conditions:
            k:{"type":"ContainersReady"}:
              .: {}
              f:lastProbeTime: {}
              f:lastTransitionTime: {}
              f:status: {}
              f:type: {}
            k:{"type":"Initialized"}:
              .: {}
              f:lastProbeTime: {}
              f:lastTransitionTime: {}
              f:status: {}
              f:type: {}
            k:{"type":"Ready"}:
              .: {}
              f:lastProbeTime: {}
              f:lastTransitionTime: {}
              f:status: {}
              f:type: {}
          f:containerStatuses: {}
          f:hostIP: {}
          f:phase: {}
          f:podIP: {}
          f:podIPs:
            .: {}
            k:{"ip":"172.30.64.142"}:
              .: {}
              f:ip: {}
          f:startTime: {}
      manager: kubelet
      operation: Update
      time: "2021-07-02T05:13:21Z"
    name: operator-76898bcfc5-rwrbc
    namespace: hypershift
    ownerReferences:
    - apiVersion: apps/v1
      blockOwnerDeletion: true
      controller: true
      kind: ReplicaSet
      name: operator-76898bcfc5
      uid: 8f74e8fa-d167-4d54-a2ce-a49ae6e379f4
    resourceVersion: "1888171"
    selfLink: /api/v1/namespaces/hypershift/pods/operator-76898bcfc5-rwrbc
    uid: 5349c63d-65c4-4a73-a4f5-065624172530
  spec:
    containers:
    - args:
      - run
      - --namespace=$(MY_NAMESPACE)
      - --deployment-name=operator
      - --metrics-addr=:9000
      command:
      - /usr/bin/hypershift-operator
      env:
      - name: MY_NAMESPACE
        valueFrom:
          fieldRef:
            apiVersion: v1
            fieldPath: metadata.namespace
      image: registry.ng.bluemix.net/armada-master/armada-hypershift-operator:8f75537383ddc03057c93e88d0bcaa694a9f03e0
      imagePullPolicy: Always
      name: operator
      ports:
      - containerPort: 9000
        name: metrics
        protocol: TCP
      resources: {}
      securityContext:
        capabilities:
          drop:
          - MKNOD
      terminationMessagePath: /dev/termination-log
      terminationMessagePolicy: File
      volumeMounts:
      - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
        name: operator-token-chq26
        readOnly: true
    dnsPolicy: ClusterFirst
    enableServiceLinks: true
    imagePullSecrets:
    - name: operator-dockercfg-rgbk6
    nodeName: 10.5.188.175
    preemptionPolicy: PreemptLowerPriority
    priority: 0
    restartPolicy: Always
    schedulerName: default-scheduler
    securityContext:
      seLinuxOptions:
        level: s0:c26,c0
    serviceAccount: operator
    serviceAccountName: operator
    terminationGracePeriodSeconds: 30
    tolerations:
    - effect: NoExecute
      key: node.kubernetes.io/not-ready
      operator: Exists
      tolerationSeconds: 300
    - effect: NoExecute
      key: node.kubernetes.io/unreachable
      operator: Exists
      tolerationSeconds: 300
    volumes:
    - name: operator-token-chq26
      secret:
        defaultMode: 420
        secretName: operator-token-chq26
  status:
    conditions:
    - lastProbeTime: null
      lastTransitionTime: "2021-07-02T05:13:14Z"
      status: "True"
      type: Initialized
    - lastProbeTime: null
      lastTransitionTime: "2021-07-02T05:13:21Z"
      status: "True"
      type: Ready
    - lastProbeTime: null
      lastTransitionTime: "2021-07-02T05:13:21Z"
      status: "True"
      type: ContainersReady
    - lastProbeTime: null
      lastTransitionTime: "2021-07-02T05:13:14Z"
      status: "True"
      type: PodScheduled
    containerStatuses:
    - containerID: cri-o://82b8b0254da0bad10fe304deeca7b6decae12837e7aeecfce9597fb9bd10ff8f
      image: registry.ng.bluemix.net/armada-master/armada-hypershift-operator:8f75537383ddc03057c93e88d0bcaa694a9f03e0
      imageID: registry.ng.bluemix.net/armada-master/armada-hypershift-operator@sha256:b1ee973567d52cb0618d573d3eee8ca069b278d07ab4875e7a1514434f060bdd
      lastState: {}
      name: operator
      ready: true
      restartCount: 0
      started: true
      state:
        running:
          startedAt: "2021-07-02T05:13:20Z"
    hostIP: 10.5.188.175
    phase: Running
    podIP: 172.30.64.142
    podIPs:
    - ip: 172.30.64.142
    qosClass: BestEffort
    startTime: "2021-07-02T05:13:14Z"
```

**Comment by relyt0925** [comment]
```
apiVersion: v1
kind: Pod
metadata:
  annotations:
    cni.projectcalico.org/podIP: 172.30.252.148/32
    cni.projectcalico.org/podIPs: 172.30.252.148/32
    k8s.v1.cni.cncf.io/network-status: |-
      [{
          "name": "",
          "ips": [
              "172.30.252.148"
          ],
          "default": true,
          "dns": {}
      }]
    k8s.v1.cni.cncf.io/networks-status: |-
      [{
          "name": "",
          "ips": [
              "172.30.252.148"
          ],
          "default": true,
          "dns": {}
      }]
    openshift.io/scc: anyuid
  creationTimestamp: "2021-07-02T05:13:25Z"
  generateName: control-plane-operator-6c77cc89d-
  labels:
    name: control-plane-operator
    pod-template-hash: 6c77cc89d
  managedFields:
  - apiVersion: v1
    fieldsType: FieldsV1
    fieldsV1:
      f:metadata:
        f:generateName: {}
        f:labels:
          .: {}
          f:name: {}
          f:pod-template-hash: {}
        f:ownerReferences:
          .: {}
          k:{"uid":"0d77ec30-fecf-48e2-96ee-56a263982434"}:
            .: {}
            f:apiVersion: {}
            f:blockOwnerDeletion: {}
            f:controller: {}
            f:kind: {}
            f:name: {}
            f:uid: {}
      f:spec:
        f:containers:
          k:{"name":"control-plane-operator"}:
            .: {}
            f:args: {}
            f:command: {}
            f:env:
              .: {}
              k:{"name":"MY_NAMESPACE"}:
                .: {}
                f:name: {}
                f:valueFrom:
                  .: {}
                  f:fieldRef:
                    .: {}
                    f:apiVersion: {}
                    f:fieldPath: {}
            f:image: {}
            f:imagePullPolicy: {}
            f:name: {}
            f:resources: {}
            f:terminationMessagePath: {}
            f:terminationMessagePolicy: {}
        f:dnsPolicy: {}
        f:enableServiceLinks: {}
        f:restartPolicy: {}
        f:schedulerName: {}
        f:securityContext:
          .: {}
          f:seLinuxOptions:
            f:level: {}
        f:serviceAccount: {}
        f:serviceAccountName: {}
        f:terminationGracePeriodSeconds: {}
    manager: kube-controller-manager
    operation: Update
    time: "2021-07-02T05:13:25Z"
  - apiVersion: v1
    fieldsType: FieldsV1
    fieldsV1:
      f:metadata:
        f:annotations:
          f:cni.projectcalico.org/podIP: {}
          f:cni.projectcalico.org/podIPs: {}
    manager: calico
    operation: Update
    time: "2021-07-02T05:13:27Z"
  - apiVersion: v1
    fieldsType: FieldsV1
    fieldsV1:
      f:metadata:
        f:annotations:
          f:k8s.v1.cni.cncf.io/network-status: {}
          f:k8s.v1.cni.cncf.io/networks-status: {}
    manager: multus
    operation: Update
    time: "2021-07-02T05:13:27Z"
  - apiVersion: v1
    fieldsType: FieldsV1
    fieldsV1:
      f:status:
        f:conditions:
          k:{"type":"ContainersReady"}:
            .: {}
            f:lastProbeTime: {}
            f:lastTransitionTime: {}
            f:status: {}
            f:type: {}
          k:{"type":"Initialized"}:
            .: {}
            f:lastProbeTime: {}
            f:lastTransitionTime: {}
            f:status: {}
            f:type: {}
          k:{"type":"Ready"}:
            .: {}
            f:lastProbeTime: {}
            f:lastTransitionTime: {}
            f:status: {}
            f:type: {}
        f:containerStatuses: {}
        f:hostIP: {}
        f:phase: {}
        f:podIP: {}
        f:podIPs:
          .: {}
          k:{"ip":"172.30.252.148"}:
            .: {}
            f:ip: {}
        f:startTime: {}
    manager: kubelet
    operation: Update
    time: "2021-07-02T05:13:35Z"
  name: control-plane-operator-6c77cc89d-hc9sc
  namespace: master-tyler12
  ownerReferences:
  - apiVersion: apps/v1
    blockOwnerDeletion: true
    controller: true
    kind: ReplicaSet
    name: control-plane-operator-6c77cc89d
    uid: 0d77ec30-fecf-48e2-96ee-56a263982434
  resourceVersion: "1888367"
  selfLink: /api/v1/namespaces/master-tyler12/pods/control-plane-operator-6c77cc89d-hc9sc
  uid: 1bcfee2f-35df-4a0a-9af7-9626b6f7ca20
spec:
  containers:
  - args:
    - run
    - --namespace
    - $(MY_NAMESPACE)
    - --deployment-name
    - control-plane-operator
    command:
    - /usr/bin/control-plane-operator
    env:
    - name: MY_NAMESPACE
      valueFrom:
        fieldRef:
          apiVersion: v1
          fieldPath: metadata.namespace
    image: registry.ng.bluemix.net/armada-master/armada-hypershift-operator:8f75537383ddc03057c93e88d0bcaa694a9f03e0
    imagePullPolicy: Always
    name: control-plane-operator
    resources: {}
    securityContext:
      capabilities:
        drop:
        - MKNOD
    terminationMessagePath: /dev/termination-log
    terminationMessagePolicy: File
    volumeMounts:
    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
      name: control-plane-operator-token-b7fkj
      readOnly: true
  dnsPolicy: ClusterFirst
  enableServiceLinks: true
  imagePullSecrets:
  - name: control-plane-operator-dockercfg-khm4w
  nodeName: 10.241.170.189
  preemptionPolicy: PreemptLowerPriority
  priority: 0
  restartPolicy: Always
  schedulerName: default-scheduler
  securityContext:
    seLinuxOptions:
      level: s0:c26,c15
  serviceAccount: control-plane-operator
  serviceAccountName: control-plane-operator
  terminationGracePeriodSeconds: 30
  tolerations:
  - effect: NoExecute
    key: node.kubernetes.io/not-ready
    operator: Exists
    tolerationSeconds: 300
  - effect: NoExecute
    key: node.kubernetes.io/unreachable
    operator: Exists
    tolerationSeconds: 300
  volumes:
  - name: control-plane-operator-token-b7fkj
    secret:
      defaultMode: 420
      secretName: control-plane-operator-token-b7fkj
status:
  conditions:
  - lastProbeTime: null
    lastTransitionTime: "2021-07-02T05:13:25Z"
    status: "True"
    type: Initialized
  - lastProbeTime: null
    lastTransitionTime: "2021-07-02T05:13:35Z"
    status: "True"
    type: Ready
  - lastProbeTime: null
    lastTransitionTime: "2021-07-02T05:13:35Z"
    status: "True"
    type: ContainersReady
  - lastProbeTime: null
    lastTransitionTime: "2021-07-02T05:13:25Z"
    status: "True"
    type: PodScheduled
  containerStatuses:
  - containerID: cri-o://20e33984177491350435742f40557d20975a600d3dc1f602172777fd541cedf5
    image: registry.ng.bluemix.net/armada-master/armada-hypershift-operator:8f75537383ddc03057c93e88d0bcaa694a9f03e0
    imageID: registry.ng.bluemix.net/armada-master/armada-hypershift-operator@sha256:b1ee973567d52cb0618d573d3eee8ca069b278d07ab4875e7a1514434f060bdd
    lastState: {}
    name: control-plane-operator
    ready: true
    restartCount: 0
    started: true
    state:
      running:
        startedAt: "2021-07-02T05:13:34Z"
  hostIP: 10.241.170.189
  phase: Running
  podIP: 172.30.252.148
  podIPs:
  - ip: 172.30.252.148
  qosClass: BestEffort
  startTime: "2021-07-02T05:13:25Z"
```

File: cmd/install/assets/hypershift_operator.go [hunk 1]
```diff
@@ -36,6 +36,11 @@ type HyperShiftOperatorDeployment struct {
 }
 
 func (o HyperShiftOperatorDeployment) Build() *appsv1.Deployment {
+	// needed since hypershift operator runs with anyuuid scc
+	nonRootUser := int64(1000)
```
Comment (enxebre) [inline cmd/install/assets/hypershift_operator.go:40]: nit: can use pointer.Int64Ptr()

Comment (relyt0925) [reply cmd/install/assets/hypershift_operator.go:40]: I can!

**Pull request merged**